src/test/java/com/synopsys/integration/alert/database/api/ProcessingJobAccessorTestIT.java

```diff
@@ -0,0 +1,240 @@
+        ConfigurationFieldModel providerConfigName = ConfigurationFieldModel.create(ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME);
+        providerConfigName.setFieldValue("My Black Duck Config");
+
+        ConfigurationFieldModel blackduckUrl = ConfigurationFieldModel.create(BlackDuckDescriptor.KEY_BLACKDUCK_URL);
+        blackduckUrl.setFieldValue("https://a-blackduck-server");
+        ConfigurationFieldModel blackduckApiKey = ConfigurationFieldModel.create(BlackDuckDescriptor.KEY_BLACKDUCK_API_KEY);
+        blackduckApiKey.setFieldValue("123456789012345678901234567890123456789012345678901234567890");
+        ConfigurationFieldModel blackduckTimeout = ConfigurationFieldModel.create(BlackDuckDescriptor.KEY_BLACKDUCK_TIMEOUT);
+        blackduckTimeout.setFieldValue("300");
+
+        List<ConfigurationFieldModel> providerConfigFields = List.of(providerConfigEnabled, providerConfigName, blackduckUrl, blackduckApiKey, blackduckTimeout);
+        providerConfigId = configurationAccessor.createConfiguration(new BlackDuckProviderKey(), ConfigContextEnum.GLOBAL, providerConfigFields).getConfigurationId();
+    }
+
+    @AfterEach
+    public void removeCreatedJobsIfExist() {
+        CREATED_JOBS.forEach(jobAccessor::deleteJob);
+        CREATED_JOBS.clear();
+        configurationAccessor.deleteConfiguration(providerConfigId);
+    }
+
+    @Test
+    public void testMatchingEnabledJobsUniqueJobsPerPage() throws InterruptedException {
+        int expectedNumOfJobs = 1000;
+        Set<UUID> previousJobIdSet = new HashSet<>();
+        createJobs(createDistributionJobModels(List.of(VulnerabilitySeverityType.LOW.name()), 100));
+        createJobs(createDistributionJobModels(List.of(VulnerabilitySeverityType.LOW.name(), VulnerabilitySeverityType.HIGH.name()), 100));
+        createJobs(createDistributionJobModels(List.of(VulnerabilitySeverityType.LOW.name()), 100));
+        createJobs(createDistributionJobModels(List.of(VulnerabilitySeverityType.LOW.name()), 100));
+        createJobs(createDistributionJobModels(List.of(VulnerabilitySeverityType.LOW.name()), 100));
+        createJobs(createDistributionJobModels(List.of(VulnerabilitySeverityType.LOW.name()), 100));
+        createJobs(createDistributionJobModels(List.of(VulnerabilitySeverityType.LOW.name()), 100));
+        createJobs(createDistributionJobModels(List.of(VulnerabilitySeverityType.LOW.name()), 100));
+        createJobs(createDistributionJobModels(List.of(VulnerabilitySeverityType.LOW.name()), 100));
+        createJobs(createDistributionJobModels(List.of(VulnerabilitySeverityType.LOW.name()), 100));
+
+        List<DetailedNotificationContent> notifications = new ArrayList<>();
+        notifications.addAll(createVulnerabilityNotificationWrappers(List.of(VulnerabilitySeverityType.LOW.name()), PROJECT_NAME_1));
+        notifications.addAll(createVulnerabilityNotificationWrappers(List.of(VulnerabilitySeverityType.HIGH.name()), PROJECT_NAME_2));
```

Comment:
createVulnerabilityNotificationWrapper take from JobNotificationMapperTestIT.  We may want to add some utilities to make it easier to create jobs and save notifications.  Just wrote this to test the issue.  We can clean it up more in the main branch.